### PR TITLE
fix RunPlugin for NodeJS

### PR DESCRIPTION
**Pull request by tgummerer**
RunPlugin for NodeJS is currently broken in two ways:
- The program name that's passed through is currently req.Info.EntryPoint, which generally is always `.`.  This doesn't work since we're passing `req.Pwd` as the pwd, so we need to pass `req.Info.ProgramDirectory` instead.
- It's not passing `req.Args` correctly, which means that the NodeJS program won't get an engine address.

Fix those and add some tests for RunPlugin.  Since we're already adding tests, add them for Go and Python as well.

### CODE REVIEW

File: sdk/nodejs/cmd/pulumi-language-nodejs/main.go [hunk 1]
```diff
@@ -1403,9 +1403,10 @@ func (host *nodeLanguageHost) RunPlugin(
 		return err
 	}
 
-	nodeargs = append(nodeargs, req.Info.EntryPoint)
```
Comment (Frassle) [inline sdk/nodejs/cmd/pulumi-language-nodejs/main.go:1406]: I don't think EntryPoint is ever anything but "." today, but if we add "main" support to PulumiPlugin.json we'd probably need to use it in here somewhere.

**Review by Frassle — APPROVED** [review]

File: tests/integration/run_plugin/provider-nodejs/index.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    async construct(name: string, type: string, inputs: pulumi.Inputs,
+        options: pulumi.ComponentResourceOptions): Promise<provider.ConstructResult> {
+	const typeName = type.split(":", 3)[2];
+        return {
+            urn: pulumi.createUrn(type, name),
+	    state: {
+		"ITS_ALIVE": "IT'S ALIVE!",
+	    }
+        }
```
Comment (julienp) [inline tests/integration/run_plugin/provider-nodejs/index.ts:17]: there's some indentation thing going on here, might not be visible depending on your GitHub settings
Screenshot 2024-11-22 at 13 21 22

File: tests/integration/integration_go_test.go [hunk 3]
```diff
@@ -1400,3 +1400,34 @@ func TestLogDebugGo(t *testing.T) {
...
+	e.RunCommand(filepath.Join("venv", "bin", "python"), "-m", "pip", "install", "-e", pythonSdkPath)
+	e.Env = append(e.Env, "PATH="+filepath.Join(e.CWD, "venv", "bin")+string(os.PathListSeparator)+os.Getenv("PATH"))
+
+	e.CWD = filepath.Join(e.RootPath, "go")
+	sdkPath, err := filepath.Abs("../../sdk/")
+	require.NoError(t, err)
+
+	e.RunCommand("go", "mod", "edit", "-replace=github.com/pulumi/pulumi/sdk/v3="+sdkPath)
+	e.RunCommand("pulumi", "stack", "init", "debugger-test")
+	e.RunCommand("pulumi", "stack", "select", "debugger-test")
```
Comment (julienp) [inline tests/integration/integration_go_test.go:1431]: copy-pasta stack name I think

File: tests/integration/run_plugin/provider-nodejs/index.ts (hunk 2 above)
Comment (tgummerer) [reply tests/integration/run_plugin/provider-nodejs/index.ts]: Ah thanks!  I usually rely on the linter to catch this, but we're not linting these.

File: tests/integration/integration_go_test.go (hunk 3 above)
Comment (tgummerer) [reply tests/integration/integration_go_test.go]: indeed.  Renamed it.

**Pull request merged**